src/SmartNumber.php

```diff
@@ -360,23 +358,24 @@ private function setValue(string $input): void
 		} catch (NumberFormatException $e) {
 		}
 
-		if (preg_match('/^(?<mantissa>-?\d*[.]?\d+)(e|E|^)(?<exponent>-?\d*[.]?\d+)$/', $input, $parseExponential)) {
```

Comment:
Where is support for numbers like `3E6` or `25.13e17`?

Reply:
sorry.

Reply:
It is supported natively by brick math (BigNumber::of())